crates/matrix-sdk/src/sliding_sync/mod.rs

```diff
@@ -415,21 +427,29 @@ impl SlidingSync {
         // Everything went well, we can update the position markers.
         //
         // Save the new position markers.
-        let mut position = self.inner.position.write().unwrap();
-        position.pos = pos;
-        position.delta_token = sliding_sync_response.delta_token.clone();
+        position_guard.pos = pos;
+        position_guard.delta_token = sliding_sync_response.delta_token.clone();
 
         // Keep this position markers in memory, in case it pops from the server.
         let mut past_positions = self.inner.past_positions.write().unwrap();
-        past_positions.push(position.clone());
+        past_positions.push(position_guard.clone());
+
+        // Release the position markers lock.
+        // It means that other request can start be sent.
+        drop(position_guard);
```

Comment:
It's not necessary _per se_ to have the `drop` here, since it will be dropped naturally at the end of the function, which happens on the next line. We could keep this if we wanted to keep the nice comment, though.

Reply:
Yeah I know it's implicit, but I prefer to get an explicit drop here so that the whole logic is clear and documented.